Added the new changes for 1.9 Pipelines RN (#55040)
openshift/openshift-docs: modules/op-release-notes-1-9.adoc

[id="operator-new-features-1-9_{context}"]
=== Operator

> * With this update, {pac} is installed by default. You can disable {pac} by using the following flag:

Author: Fixed.

> * With this update, {pac} is installed by default. You can disable {pac} by using the following flag:

Reviewer: Another possibility, if you encounter a similar construction in the future: 
```
To disable {pac}, enter:
```

> * With this update, {pac} is installed by default and it is not a part of `TektonAddon`. You can disable {pac} by using the following flag:

Reviewer 2: I am thinking we should avoid mentioning about `TektonAddon`  as it may confuse to new PAC user 

@vdemeester wdyt for below sentence 
```suggestion
* With this update, {pac} is installed by default. You can disable {pac} by using the following flag:
```

Reviewer 3: @dhruv, please see the guidance wrt Kubernetes Objects here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references especially the second note and the examples above it. That is relevant if you retain the words - `TektonAddon` If not please read and apply when you use CRDs.

Reviewer 4: Wrong dev

Author: @Preeticp - I have decided to remove the words TektonAddon as instructed by the SME for the time being.

+
[source,yaml]

----

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer: I believe the extra blank lines break the alignment of the command with the bullet point. To catch formatting/layout issues, I recommend examining the preview of your work. We have some PR review checklists that help us walk through all the correct steps. @bburt-rh has a nice easy one. Brian, can  you share the link here?

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer 5: @rolfedh @Dhruv-Soni11 - I added the link to my checklist in the Slack review request for this PR

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer 5: ```suggestion
$ oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'
```

- Also the CLI command needs to have a prompt that indicates the proper level of permissions to run it.

Author: Fixed.

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Author: Fixed this comment in the latest iteration. Kindly check and confirm if it is fine.

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer: @dhruv Based on your examination of the preview, you can mark this as resolved if the formatting looks good now.

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer: LGTM - needs SME review @vdemeester

> oc patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Author: Hi @rolfedh - The content looks fine to me in the preview. Adding a screenshot for our reference. 
![Screenshot from 2023-02-08 17-03-34](https://user-images.githubusercontent.com/108407409/217518813-8a7a7883-89ab-43fc-9fb2-1c512eabb921.png)

> kubectl patch tektonconfig config --type="merge" -p '{"spec": {"platforms": {"openshift":{"pipelinesAsCode": {"enable": false}}}}}'

Reviewer 6: @Dhruv-Soni11 In OpenShift doc, we have to use `oc` command

Reviewer 2: I see everywhere its used `oc` in Openshift doc but ya will defer to @sounix000 input

Reviewer 3: +1 to @jkandasa's comment. Please change any instance of `kubectl` to `oc`

Author: @Preeticp - I see from @sounix000's status is OOO till 29th Jan. Can we ask some other writer or should I go ahead and change any instances of `kubectl` to `oc` in this command?

Author: Thank you for informing me about this change, Preeti. I have gone ahead and replaced an instance of `kubectl` to `oc` in the content. @VeereshAradhya @ppitonak and @rupalibehera, could you confirm if this command is accurate after this change.

* With this update, support for `tekton.dev/v1alpha1` API pipeline resources, such as Pipeline, PipelineRun, Task, Clustertask, and TaskRun has been removed.	

> * With this update, the `tkn-pac setup` command has been removed. You can now re-use the functionality provided by the `tkn-pac setup` command by using the `tkn-pac webhook add` and `tkn-pac webhook update-token` commands. You can use the `tkn-pac webhook add` command to re-add a webhook to an existing Git repository. You can use the `tkn-pac webhook update-token` command to update the personal provider access token for an existing Secret object in the Git repository.

Reviewer: Maybe, for minimalism:
```suggestion
* With this update, the `tkn-pac setup` command has been removed. Instead, use the `tkn-pac webhook add` command to re-add a webhook to an existing Git repository. And use the `tkn-pac webhook update-token` command to update the personal provider access token for an existing Secret object in the Git repository.
```

Author: Fixed.

> * With this update, a namespace that runs a pipeline with default settings does not apply the `pod-security.kubernetes.io/enforce:privileged` label to a workload.

Author: @rupalibehera  @ppitonak - I have added the inputs as per your feedback. Kindly have a look and let me know your comments, if any.

Reviewer 7: @vdemeester could you please check if this looks correct

Reviewer 8: I think that is correct, not sure exactly how is the wording around this in the rest of the openshift docs, but it seems alright there.

* Running the `tkn pac resolve -f <filename> | oc create -f` command may not provide expected results, if the `tkn pac resolve` command uses a templated parameter value to function.	
+	

> Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> -o tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command. For example, `tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && oc create -f /tmp/pull-request-resolved.yaml`.	

Reviewer 9: should be fine! 👍🏻

> Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> -o tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command. For example, `tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && oc create -f /tmp/pull-request-resolved.yaml`.	

Author: @VeereshAradhya @rupalibehera @ppitonak - Just updating this thread with a gentle reminder for your inputs. PTAL.

> Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> -o tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command. For example, `tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && oc create -f /tmp/pull-request-resolved.yaml`.	

Reviewer 7: looks good

// Vincent Demeester	

* Before this update, webhook validation was removed on deletion of old versions of invalid objects. With this update, this issue is resolved.	
// Vincent Demeester @vdeemester

> * Before this update, you could not set some fields of `timeouts`. With this update, this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Author: Fixed as per your suggestion @rolfedh. PTAL.

> * Before this update, you could not set some fields of `timeouts`. With this update, this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Reviewer: LGTM - needs SME review @vdemeester

> * Before this update, you could not set some fields of `timeouts`. With this update, this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Reviewer 8: Yes 👍🏼

* Before this update, if Pipeline timeout was specified as zero `(i.e. timeouts.pipeline = 0)`, a user could not specify a value for other timeout fields. This update resolves the issue. Now, the PipelineRun validation has been updated and you can specify a value for the `tasks` timeout field. If you do not set a value for these fields, they can use default values.

> // Vincent Demeester @vdeemester

Author: Yes, this would be duplicate content. I am going to delete this sentence.

// Vincent Demeester	

* Before this update, webhook validation was removed on deletion of old versions of invalid objects. With this update, this issue is resolved.	
// Vincent Demeester @vdeemester

> * Before this update, you could not set some fields of `timeouts`. With this update, this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Reviewer: If I were a user, I might have some questions when I read this RN:
- Where or how do I find these fields you're talking about?
- Which fields were previously unsettable and are now settable?
- Are **some** fields still unsettable?  
- Where can I see the default values?

Maybe something more like this:
```suggestion
* Before this update, you could not set some `timeouts` fields in the <name of resource or config file>. This update resolves the issue. Now, you can set the values of the following `timeouts` fields: <field a>, <field b>, and <field c>. If you don't set a value for these fields, they use default values. To learn more, see: link:<URL etc.>
```

Author: @vdemeester @rupalibehera - Could you guys help me with some inputs here please?

Reviewer 8: The fix is https://github.com/tektoncd/pipeline/pull/5460 .
> This commit updates PipelineRun validation to allow a user to specify only a tasks timeout or only a finally timeout.
Prior to this commit, if a user wanted to specify no Pipeline timeout (i.e. timeouts.pipeline = 0), they would not be able to specify values for the other timeouts fields.

This is on PipelineRun indeed 👼🏼

Author: Thanks for providing feedback folks. I have updated the content. PTAL. Thanks.

Reviewer: Outdated. See following comment.
